### PR TITLE
feat: add support for m1 macs

### LINKED ISSUE #4: Move this line after brew install
Having this before installing brew just causes an error. 🤦‍♂️ Thanks @richard-rance for catching this

https://github.com/vendasta/setup-new-computer-script/blob/77aad569ea64eb1274e1d5dda4240be4d21554f9/setup-new-computer.sh#L335-L337

### LINKED ISSUE #11: Support M1 Macs
New macs using the Apple Silicon processors need a different `brew` install and initiation to work, as files are stored in a new location.

### PR DESCRIPTION
**Pull request by joelkesler**
### Works on both Intel and M1/M2 Macs

Fixes: https://github.com/vendasta/setup-new-computer-script/issues/4 -  M1 Macs require different `brew` paths. On Intel, homebrew is installed in `/opt/homebrew/bin` white on M1/M2 computers, homebrew is now installed in `/usr/local/bin`. The script now check what processor you are running and adjusts the path to match. 

The script will also modify`.zprofile` and `.bash_profile` to start homebrew from the proper path depending on your processor. 
<br>

### Skip installation of main apps if they already exists.

Fixes: https://github.com/vendasta/setup-new-computer-script/issues/11 - The script will now check if apps such as Google Chrome or Slack is already installed and will skip the download if they do. Previously, the script would download the app first even if the app was installed. This speeds up the script.
<br>

### Give explanation why Git with Github may fail with errors if you try to clone a repo before adding a generated SSH key to your Github account 
After the install script finishes, Git is set to connect to github via ssh instead of the https. If you do not generate and add an ssh key to your github account, you will get errors when connecting to github until you do. This will help explain why this happens.

### CODE REVIEW

**Comment by joelkesler** [comment]
```
CPU=$(uname -p)
if [[ "$CPU" == "arm" ]]; then
    echo "Its arm"
else
    echo "Its intel"
fi


if [[ "$(uname -p)" == "arm" ]]; then
    echo "Its arm"
else
    echo "Its intel"
fi
```

File: README.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
+# Temp
+
+```
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/vendasta/setup-new-computer-script/jk/add-support-for-m1-macs/setup-new-computer.sh)"
+```
+
```
Comment (joelkesler) [inline README.md]: remove before merging

File: README.md [hunk 2]
```diff
@@ -37,7 +49,7 @@ Afer you have run the script, please complete the following steps to finish sett
 
    
 1. **Github Command-line SSH Authentication**\
-   Do the following to authorize Github on your computer:
+   Git is now configured to use SSH by default for github urls. You will need to generate and add an SSH key to your Github account or you will run into errors. Do the following to authorize Github on your computer:
```
Comment (joelkesler) [inline README.md:46]: add a comment, and a link on how to disable this

File: README.md [hunk 3]
```diff
@@ -696,16 +724,10 @@ The following examples were helpful in building this script.
...
-  `/opt/homebrew/bin`
+## Tips for Testing in Virtual Machines
 
-- More reading:
-  - https://www.wisdomgeek.com/development/installing-intel-based-packages-using-homebrew-on-the-m1-mac/
-  - https://github.com/Homebrew/discussions/discussions/417
+- Intel - Install Intel Mac OS VMs with Virtualbox
+  https://github.com/myspaghetti/macos-virtualbox
+- M1/M2 - Install UTM to run Mac OS VMs using Apple's Virtualization.framework
+  https://mac.getutm.app/
```
Comment (joelkesler) [inline README.md]: write more about testing

File: setup-new-computer.sh [hunk 4]
```diff
@@ -261,10 +272,6 @@ EOT
 }
 
 
-# Get root user for later. Brew needs the user to be admin to install 
-sudo ls > /dev/null
-
-
```
Comment (joelkesler) [inline setup-new-computer.sh:267]: This didn't do much. Brew still needed a password. removing

**Comment by jbarth-va** [comment]
I haven't used the new computer script yet, but seems like it will make life easier if I were to get my hands on a shiny new work mac. 

@vendasta/ioscotw

**Pull request merged**